app/src/org/commcare/activities/connect/ConnectActivity.java

```diff
@@ -147,4 +232,47 @@ public void startAppValidation() {
         i.putExtra(CommCareVerificationActivity.KEY_LAUNCH_FROM_SETTINGS, true);
         verificationLauncher.launch(i);
     }
+
+    public void getJobDetails() {
+        ApiConnect.getConnectOpportunities(ConnectActivity.this, new IApiCallback() {
+            @Override
+            public void processSuccess(int responseCode, InputStream responseData) {
+                try {
+                    String responseAsString = new String(StreamsUtil.inputStreamToByteArray(responseData));
+                    if (responseAsString.length() > 0) {
+                        //Parse the JSON
+                        JSONArray json = new JSONArray(responseAsString);
+                        List<ConnectJobRecord> jobs = new ArrayList<>(json.length());
```

Comment:
We should save the new jobs list to the DB since we just got the most updated info. See ConnectDatabaseHelper.storeJobs, you can set pruneMissing=true in this case.

Reply:
Done

Reply:
Oops! Close, but the current code stores the "jobs" list while it's still blank! The call to storeJobs() needs to happen after the for loop that parsed the JSON into the list

Reply:
Okay I will set storeJobs() after for the loop